### PR TITLE
LRDOCS-4611

### PR DESCRIPTION
**Pull request by akosthurzo**
Hi Cody,

I am sending the info that could be used in how-tos or tutorials about creating staged model repositories. If you have any questions or something needs clarification, please let me know.

Thanks,
Akos

/cc @matethurzo

### CODE REVIEW

File: modules/apps/web-experience/export-import/export-import-api/src/main/java/com/liferay/exportimport/staged/model/repository/StagedModelRepository.java [hunk 1]
```diff
@@ -49,16 +84,32 @@ public default T fetchMissingReference(String uuid, long groupId) {
 	public List<T> fetchStagedModelsByUuidAndCompanyId(
 		String uuid, long companyId);
 
+	/**
```
Comment (codyhoag) [inline modules/apps/web-experience/export-import/export-import-api/src/main/java/com/liferay/exportimport/staged/model/repository/StagedModelRepository.java]: @akosthurzo can you add what this particular method (and the other methods that you've done this way) does? You redirect to other methods that developers should use, but we don't have a good understanding of what this interface provides for those methods. We can keep the recommended methods there, but just state them after the initial description of the method.

File: modules/apps/web-experience/export-import/export-import-api/src/main/java/com/liferay/exportimport/staged/model/repository/StagedModelRepository.java [hunk 2]
```diff
@@ -49,16 +84,32 @@ public default T fetchMissingReference(String uuid, long groupId) {
...
+	 * If possible, use return _yourEntityLocalService.getExportActionableDynamicQuery(portletDataContext); as the implementation.
+	 * @param portletDataContext
+	 * @return
```
Comment (codyhoag) [inline modules/apps/web-experience/export-import/export-import-api/src/main/java/com/liferay/exportimport/staged/model/repository/StagedModelRepository.java]: @akosthurzo can you also add the info for all `@param` and `@return` tags in this new Javadoc? Most of them are probably pretty self-explanatory, but having descriptions for them removes any doubt readers may have.

**Comment by codyhoag** [comment]
Thanks @akosthurzo. Can you answer one question I have on the `StagedModelRepository` framework before I start the review?

Can you explain the differences between using the `StagedModelRepository` framework over local services? Our tutorials (e.g., [this one](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-1/developing-portlet-data-handlers)) show using local services *instead* of the `StagedModelRepository` framework. What's pros/cons for both? This will help for the tutorial I write (and maybe add more info to existing Javadoc).

Thanks! 😊

**Comment by matethurzo** [comment]
@codyhoag the primary problem with out services that they don't share a common ancestor (interface or base class), and because of that it's impossible for the staging framework to call those methods. When the staging framework is operating on the entities, we don't have the information about the entity's local services, so we needed a framework where we can link the entities to a local service on a code level as the StagedModelRepositories are registered to a specific entity.
The repositories are specialized local services to the staging needs.

they creates the portal entity (StagedModel) -> StagedModelRepository -> LocalService linking, and the staging framework will able able to call them independently from the fact that which entity is currently processed.

**Comment by codyhoag** [comment]
@akosthurzo I made a couple in-line comments on this PR about the Javadoc. I wasn't sure if those fell off the radar. Let me know if you have questions about those requests.

**Comment by akosthurzo** [comment]
Hi @codyhoag,

Sorry, I was OOO. I have added more description, if you need more info, please let me know.

Thanks,
Akos

**Comment by codyhoag** [comment]
Sent akosthurzo#78 for review of changes.

**Pull request closed without merging**